### PR TITLE
feat: Add Marinade SDK as a vault

### PR DESCRIPTION
**Pull request by ayushpaharia**
Worked on adding **Marinande Finance** as a Vault in Navigator. Currently I have implemented layouts, ids and types would like somehelp in the implementation on tests and infos

A summary of my changes are as follows:
- Added a folder called marinade to the src directory.
- Added marinade implementation inside said folder.
    
@wei1769 @ironaddicteddog

### CODE REVIEW

File: src/marinade/layouts.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    publicKey("operationalSolAccount"),
+    publicKey("treasuryMsolAccount"),
+    u8("reserveBumpSeed"),
+    u8("msolMintAuthorityBumpSeed"),
+    u64("rentExemptForTokenAcc"),
+    struct([u32("basisPoints")], "rewardFee"),
+    struct(
+      [
+        struct(
+          [[publicKey("account"), u32("itemSize"), u32("count"), publicKey("newAccount"), u32("copiedCount")]],
```
Comment (wei1769) [inline src/marinade/layouts.ts:16]: The array is wrapped inside another array
Try remove the extra `[`

File: src/marinade/layouts.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        u64("lastStakeDeltaEpoch"),
+        u64("minStake"),
+        u32("extraStakeDeltaRuns"),
+      ],
+      "stakeSystem"
+    ),
+    struct(
+      [
+        struct(
+          [[publicKey("account"), u32("itemSize"), u32("count"), publicKey("newAccount"), u32("copiedCount")]],
```
Comment (wei1769) [inline src/marinade/layouts.ts:32]: The array is wrapped inside another array
Try remove the extra `[`

**Comment by ayushpaharia** [comment]
Added Test for marinade finance vault in `test/marinade.ts`
Cleaned up the `MARINADE_FINANCE_ACCOUNT_STATE` layout into smaller ones.

File: src/marinade/infos.ts [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    //     publicKey("adminAuthority"),
+    //     publicKey("operationalSolAccount"),
+    //     publicKey("treasuryMsolAccount"),
+    //     u8("reserveBumpSeed"),
+    //   ],
+    //   "State"
+    // );
+
+    const decodeData = MARINADE_FINANCE_ACCOUNT_STATE.decode(data);
+    console.log({ decodeData });
```
Comment (wei1769) [inline src/marinade/infos.ts:56]: Plz check all data type is correct 
Some properties type are `number` instead of `BN`.

Comment (wei1769) [reply src/marinade/infos.ts:56]: Also plz remove debug message after done fixing

File: src/marinade/infos.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+    _programId: PublicKey = MARINADE_PROGRAM_ID
+  ): PublicKey {
+    return PublicKey.findProgramAddressSync(
+      [userKey.toBuffer(), TOKEN_PROGRAM_ID.toBuffer(), MSOL_MINT_ADDRESS.toBuffer()],
+      ASSOCIATED_TOKEN_PROGRAM_ID
+    )[0];
+  }
+
+  static parseDepositor(data: Buffer, depositorId: PublicKey): types.DepositorInfo {
+    const decodeData = MARINADE_FINANCE_ACCOUNT_STATE.decode(data);
```
Comment (wei1769) [inline src/marinade/infos.ts:140]: We are using Token account as depositor here.
The layout here is wrong

**Comment by wei1769** [comment]
There's still Withdrawer haven't implemented?

**Comment by wei1769** [comment]
Try not to use merge 
Use rebase instead

File: src/marinade/infos.ts (hunk 4 above)
Comment (ayushpaharia) [reply src/marinade/infos.ts:126]: Can you clarify?

**Comment by ayushpaharia** [comment]
> There's still Withdrawer haven't implemented?

I need some clarification from @wei1769 on that

**Comment by ironaddicteddog** [comment]
Move to #110

**Pull request closed without merging**